pkg/backup/backup_job.go

```diff
@@ -611,114 +582,60 @@ func (b *backupResumer) Resume(ctx context.Context, execCtx interface{}) error {
 		// move before we resolve the destination, we guarantee that we push the old
 		// chain.
 		insqlDB := p.ExecCfg().InternalDB
-		if err := insqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
-			return planSchedulePTSChaining(ctx, p.ExecCfg().JobsKnobs(), txn, &details, b.job.CreatedBy())
+
+		// Backup compactions do not read keys from the database, so do not
+		// need to lay a PTS.
+		if err = insqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+			return planSchedulePTSChaining(ctx, p.ExecCfg().JobsKnobs(), txn, &initialDetails, b.job.CreatedBy())
 		}); err != nil {
 			return err
 		}
-
-		var err error
-		backupDest, err = backupdest.ResolveDest(ctx, p.User(), details.Destination, details.EndTime, p.ExecCfg())
+		backupDest, err = backupdest.ResolveDest(
+			ctx, p.User(), initialDetails.Destination, initialDetails.EndTime, p.ExecCfg(),
+		)
 		if err != nil {
 			return err
 		}
-		defaultURI = backupDest.DefaultURI
-	}
-
-	// The backup job needs to lay claim to the bucket it is writing to, to
-	// prevent concurrent backups from writing to the same location.
-	//
-	// If we have already locked the location, either on a previous resume of the
-	// job or during planning because `clusterversion.BackupResolutionInJob` isn't
-	// active, we do not want to lock it again.
-	foundLockFile, err := backupinfo.CheckForBackupLock(ctx, p.ExecCfg(), defaultURI, b.job.ID(), p.User())
-	if err != nil {
-		return err
 	}
 
 	// TODO(ssd): If we restricted how old a resumed job could be,
 	// we could remove the check for details.URI == "". This is
-	// present to guard against the case where we have already
-	// written a BACKUP-LOCK file during planning and do not want
-	// to re-check and re-write the lock file. In that case
-	// `details.URI` will non-empty.
-	if details.URI == "" && !foundLockFile {
-		if err := backupinfo.CheckForPreviousBackup(ctx, p.ExecCfg(), backupDest.DefaultURI, b.job.ID(),
-			p.User()); err != nil {
-			return err
-		}
-
-		if err := p.ExecCfg().JobRegistry.CheckPausepoint("backup.before.write_lock"); err != nil {
-			return err
-		}
-
-		if err := backupinfo.WriteBackupLock(ctx, p.ExecCfg(), backupDest.DefaultURI,
-			b.job.ID(), p.User()); err != nil {
-			return err
-		}
-
-		if err := p.ExecCfg().JobRegistry.CheckPausepoint("backup.after.write_lock"); err != nil {
+	// present to guard against the case where a pre-BACKUP_LOCK cluster
+	// started a backup job, and then that job was resumed post-BACKUP_LOCK.
+	// No lock file will be found, but CheckForPreviousBackup will
+	// prevent the job from running. In that case, details.URI will be
+	// non-empty.
+	if needsInitializing {
```

Comment:
instead of having 3 consecutive `if needsInitializing` blocks, could we have just have one `if details.URI`? If you did that, there would be no need to declare `backupDetails` outside the `if detailsURI == nil` block